File: conformance/tests/if-without-else/if-without-else.runar.rb
require 'runar'

class IfWithoutElse < Runar::SmartContract
  prop :threshold, Bigint

  def initialize(threshold)
    super(threshold)
    @threshold = threshold
  end

  runar_public a: Bigint, b: Bigint
  def check(a, b)
    count = 0
    if a > @threshold
      count = count + 1
    end
    if b > @threshold
      count = count + 1
    end
    assert count > 0
  end
end

File: conformance/tests/if-without-else/if-without-else.runar.zig
const runar = @import("runar");

pub const IfWithoutElse = struct {
    pub const Contract = runar.SmartContract;

    threshold: i64,

    pub fn init(threshold: i64) IfWithoutElse {
        return .{ .threshold = threshold };
    }

    pub fn check(self: *const IfWithoutElse, a: i64, b: i64) void {
        var count: i64 = 0;
        if (a > self.threshold) {
            count = count + 1;
        }
        if (b > self.threshold) {
            count = count + 1;
        }
        runar.assert(count > 0);
    }
};
